components/tools/OmeroPy/src/omero/testlib/__init__.py

```diff
@@ -1139,8 +1141,9 @@ def wait_for_pyramid(self, id):
                     store.setPixelsId(id, True)
                     # No exception. The pyramid is now ready
                     not_ready = False
-                except Exception:
+                except Exception, ex:
                     # try again in elapse_time
+                    print count, "Pyramid not ready:", ex.message
```

Comment:
Could you log the error instead of ``print``?

Reply:
If anyone can tell me how to see the log (where it is logged etc) then that would be great.
I couldn't find where this is logged at-all at the time. Tried grepping in dist/var/log/ but didn't find anything.
The print works perfectly in that the stderr is shown in the console ONLY when the test fails.
We use 'print' in several other places in tests for this reason.

Reply:
I can remove the ```print``` but I'm not sure what the problem with ```print``` is?
We use it in a bunch of other places already since it provides useful feedback when tests fail.

```
$ grep print -r components/tools/OmeroWeb/test/
components/tools/OmeroWeb/test//integration/test_api_containers.py:        print 'well_plates_url', well_plates_url
components/tools/OmeroWeb/test//integration/test_chgrp.py:                    print o['report']
components/tools/OmeroWeb/test//integration/test_chgrp.py:                    print o['report']
components/tools/OmeroWeb/test//integration/test_containers.py:        print dir(self)
components/tools/OmeroWeb/test//integration/test_containers.py:        print img
components/tools/OmeroWeb/test//integration/test_plategrid.py:    print "WellIds", wellIds
components/tools/OmeroWeb/test//integration/test_plategrid.py:        print rspJson
components/tools/OmeroWeb/test//integration/test_show.py:        print paths
components/tools/OmeroWeb/test//integration/test_show.py:        print paths
components/tools/OmeroWeb/test//integration/test_tree.py:    #         print x
components/tools/OmeroWeb/test//integration/test_tree.py:    #         print x
components/tools/OmeroWeb/test//integration/test_tree_annotations.py:    print "annotate_project", ctx
components/tools/OmeroWeb/test//integration/test_tree_annotations.py:        print "GroupA", a.id.val
components/tools/OmeroWeb/test//integration/test_tree_annotations.py:        print "Group B", b.id.val
components/tools/OmeroWeb/test//integration/test_tree_annotations.py:        print "USER A", user.id.val, 'groupA', groupA.id.val
components/tools/OmeroWeb/test//integration/test_tree_annotations.py:        print "USER B", userb.id.val, 'groupA', groupA.id.val
components/tools/OmeroWeb/test//unit/test_render_response.py:QUERY_STRING = " %s" % string.printable
components/tools/OmeroWeb/test//unit/test_tree.py:        print marshaled
components/tools/OmeroWeb/test//unit/test_tree.py:        print expected
```

Reply:
for now we will have to go with print.
We can do further investigation cf. logging